Approximate_convex_decomposition/examples/Approximate_convex_decomposition/approx_convex_decomposition_example.cpp

```diff
@@ -0,0 +1,55 @@
+#include <CGAL/approx_decomposition.h>
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <iostream>
+#include <fstream>
+
+typedef CGAL::Simple_cartesian<double> Kernel;
+typedef CGAL::Polyhedron_3<Kernel>     Polyhedron;
+
+int main()
+{
+    // read mesh
+    Polyhedron mesh;
+    
+    std::ifstream input("data/elephant.off");
+    
+    if (!input || !(input >> mesh))
+    {
+        std::cout << "Failed to read mesh" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    if (CGAL::is_empty(mesh) || !CGAL::is_triangle_mesh(mesh))
+    {
+        std::cout << "Input mesh is invalid" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // create property map for cluster-ids
+    typedef std::map<Polyhedron::Facet_const_handle, int> Facet_int_map;
+    Facet_int_map facet_map;
+    boost::associative_property_map<Facet_int_map> facet_property_map(facet_map);
```

Comment:
for polyhedron you can use dynamic properties. See the [CGAL BGL package](https://doc.cgal.org/latest/BGL/index.html#title250.